packages/plugin-dts/src/dts.ts

```diff
@@ -38,7 +38,7 @@ export async function generateDts(data: DtsGenOptions): Promise<void> {
       declarationDir!,
       relativePath,
       basename(entrySourcePath),
-    ).replace(/\.(m?js|jsx?|m?ts|tsx?|c?js)$/, '.d.ts');
+    ).replace(/\.(js|mjs|jsx|ts|mts|tsx|cjs|cts)$/, '.d.ts');
```

Comment:
```suggestion
    ).replace(/\.(js|mjs|jsx|ts|mts|tsx|cjs|cts|cjsx|ctsx|mjsx|mtsx)$/, '.d.ts');
```

https://www.totaltypescript.com/concepts/mjs-cjs-mts-and-cts-extensions

Reply:
😂